src/effects/backends/audiounit/audiounitbackend.mm

```diff
@@ -59,51 +62,80 @@ bool canInstantiateEffect(const QString& effectId) const override {
 
         // Create a query for audio components
         AudioComponentDescription description = {
-                .componentType = kAudioUnitType_Effect,
+                .componentType = componentType,
                 .componentSubType = 0,
                 .componentManufacturer = 0,
                 .componentFlags = 0,
                 .componentFlagsMask = 0,
         };
 
         // Find the audio units
-        // TODO: Should we perform this asynchronously (e.g. using Qt's
-        // threading or GCD)?
         auto manager =
                 [AVAudioUnitComponentManager sharedAudioUnitComponentManager];
         auto components = [manager componentsMatchingDescription:description];
 
-        // Assign ids to the components
-        NSMutableDictionary<NSString*, AVAudioUnitComponent*>* componentsById =
-                [[NSMutableDictionary alloc] init];
-        QHash<QString, EffectManifestPointer> manifestsById;
+        // Load component manifests (parameters etc.) concurrently since this
+        // requires instantiating the corresponding Audio Units. We use Grand
+        // Central Dispatch (GCD) for this instead of Qt's threading facilities
+        // since GCD is a bit more lightweight and generally preferred for
+        // Apple API-related stuff.
+
+        dispatch_group_t group = dispatch_group_create();
 
         for (AVAudioUnitComponent* component in components) {
             qDebug() << "Found audio unit" << [component name];
 
             QString effectId = QString::fromNSString(
                     [NSString stringWithFormat:@"%@~%@~%@",
-                              [component manufacturerName],
-                              [component name],
-                              [component versionString]]);
-            componentsById[effectId.toNSString()] = component;
-            manifestsById[effectId] = EffectManifestPointer(
-                    new AudioUnitManifest(effectId, component));
+                            [component manufacturerName],
+                            [component name],
+                            [component versionString]]);
+
+            // Register component
+            m_componentsById[effectId.toNSString()] = component;
+
+            // Use a concurrent background GCD queue to load manifest
+            dispatch_queue_t queue = dispatch_get_global_queue(
+                    DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
+
+            dispatch_group_async(group, queue, ^{
+                // Load manifest (potentially slow blocking operation)
+                auto manifest = EffectManifestPointer(
+                        new AudioUnitManifest(effectId, component));
+
+                // Register manifest
+                auto locker = lockMutex(&m_mutex);
+                m_manifestsById[effectId] = manifest;
+            });
         }
 
-        m_componentsById = componentsById;
-        m_manifestsById = manifestsById;
+        int64_t timeoutMs = 10000;
```

Comment:
This seems quite long for a timeout. Have you found that loading the manifests takes this long?

Reply:
Not usually, I figured being a bit conservative here might be useful for slower machines, but I'd definitely be open to lower this.

I would, however, still make this timeout (which is for all plugins) a bit higher than the other one, in case GCD for some reason exhausted its thread pool, e.g. if the user has a lot (> 64) of plugins installed.